### DISCUSSION TITLE
Oqtane Documentation Contributors Meeting

### DISCUSSION THREAD

**Discussion by thabaum** [oqtane/oqtane.docs#22] [Ideas]
I am wondering if it would be a good idea to have a late night for me, early morning for you "Documentation" meeting that maybe is on the flip side of the clock to the developer meeting and at the end of the month/ same day or whatever works best for @iJungleboy as our documentation lead.

It can be scheduled for 1 hour but maybe only takes half hour if that gets all agenda accomplished.

I think it would be a great place to help explain how to contribute, what we need and share ideas on improvements.   It can also serve as another meeting a developer that may not have been able to attend developer meetings due to time.  I would be able and willing to show up to 12 hours later time to the developers meeting on any day I don't have something come up.

This can be monthly, quarterly, semi-annual, annual or once per release scheduled out of the blue but with time to know about and prepare for. I just feel if there are any issues with contributors that need to get some things addressed but can't make it to a developer meeting it would give a place to catch up asking questions and documenting solutions to the answers!  We can probably work on it on the fly more as well resolving any issues that we can provide a quick solution for once a discussion makes it clear how to better serve that area.

I think a documentation meeting prior to the developer meeting would benefit getting issues discussed at the developer meeting, having the documentation meeting after the developer meeting would help provide a chance to help inform those who didn't make the developer meeting.

I believe some people are going to want to make their application from Oqtane, not as developers of modules, but more as users of modules developers produced both as products and services.  Which is what the Oqtane community marketplace developer ecosystem is after in the long run.  Creating information to help the end-users and developers supporting those users is the ultimate goal.

Not everyone that wants to use Oqtane will be a developer is what I am saying and I don't think we should limit Oqtane to only developers because that limits the Marketplace potential and attraction.  Some are more administrative types that want solutions developed and may choose their development team to use Oqtane based on how much is available to pull from.

This meeting can be on Discord for example has been mentioned to create a channel for this or on Teams like the developer meetings.  Maybe we can use that same room with a different password?

Some more food for thought on this one in the event you feel we are ready for this next step!  I think it can be more inclusive to anyone that wants to attend instead of only developers as a last note.  Ask everyone interested to use Oqtane.org contact form and have those requests forwarded to appropriate people.  

Maybe we can have a category on the form on that page to select to help direct messages immediately.  One for developer meetings request, documentation meeting requests and so on.  Might be good to show we have meetings to new prospects letting them know this project is active in these areas and create an events page with contact forms to fill out for each.

Just thinking out loud here a little, excited to see the new discussions forums!  Hopefully this first idea hits home, but I understand if it is a bit much at the same time.  I feel if this meeting needs canceled it is acceptable, as in not a mission critical part of Oqtane Framework.  No need to lose sleep over it... but I will for the meeting.

**Comment by GregStevenson** [comment]
My normal hours are 0300-1900PST but can tweak them if needed to make this meeting. --Greg

**Comment by thabaum** [comment]
I put another post up for another type of potential meeting we could all gear more up for if we all can be on board with it.  But for now until those come to light we can still move forward on this project as we can to help keep it current as possible and a future reference for any other documentation resources created down the road.

It is great to have a resource to more directly discuss these types of issues, thank you @iJungleboy for opening up this discussion board and look forward to hearing any feedback relating when you get the time.

**Comment by iJungleboy** [comment]
I think it's a great idea.
For me the time-slot for the main Oqtane meeting is quite good, so another rotating Wednesday or something would be good for me.

**Comment by GregStevenson** [comment]
I can schedule a zoom meeting for Wed, Jan 24th at 12:00-1:00 PM ET if that works for everyone.

**Comment by thabaum** [comment]
I am fairly flexible, whatever works best for everyone else I will make work.

**Comment by iJungleboy** [comment]
@GregStevenson would an hour earlier work as well?

So Wednesday ET 11:00-12:00 (which is CET/Swiss Time 17:00-18:00)

I recommend we standardize on Teams, since just about everybody in the MS Ecosystem has this ;)

Anybody interested could mail me on info 2sxc org and I would maintain a list of people which we would invite every month.

Is that ok?

**Comment by GregStevenson** [comment]
Here is a [meeting link](https://teams.live.com/meet/9330896152316?p=hvO7rdM414PeAuDr):

I only have MS 365 Family, which does not seem to have a way of creating an online meeting separately, But it does seem to offer "join online" I set it up to occur monthly on the fourth (not last) Wednesday.

**Comment by GregStevenson** [comment]
Reminder that the meeting is tomorrow at 17:00 CET, 11:00 EST, and 8:00 PST.

**Reply by thabaum** [reply]
@GregStevenson when you have a moment, can you see if the latest dev branch fixes your upload issue?  I left a note, we may have resolved this issue you experienced as well. Saving site settings caused a file upload bug which has now been identified and resolved.   We are hoping it resolves your issue as well.
Thank you.

**Comment by thabaum** [comment]
@GregStevenson I will do my best to be there on time, thank you!

**Comment by iJungleboy** [comment]
Just to be sure we're all on the same page:

Docs meeting today at 11:00 EST or 17:00 CET / Swiss Time.

Since I didn't get @GregStevenson e-mail we'll start with his invite-link, if anything fails I'll spontaneously send a MS-Teams invite. 

@GregStevenson pls still send me an email to info 2sxc org so I have that as a backup.

**Comment by iJungleboy** [comment]
@thabaum & @GregStevenson 

I just updated the docs with a lot of stuff

1. Project overview from notes yesterday https://docs.oqtane.org/guides/docs/project.html
2. Conventions such as where to place assets https://docs.oqtane.org/guides/docs/conventions.html
3. Various productivity tips
4. created my first mermaid chart ;) https://docs.oqtane.org/guides/docs/index.html 
5. Improved the VS-Code project to auto-recommend installing plugins when opened

**Reply by thabaum** [reply]
@iJungleboy Thank you for your time and effort!  This is awesome and will help us tremendously. 🚀 

I also want to mention that if anyone sees me post some screenshots or information here or in the framework repository issues and discussions that helps you create a contribution to Oqtane documentation you have my permission to utilize them as needed to generate the assets and information.  Anything to help with progress here I am 100% for!

**Reply by thabaum** [reply]
@iJungleboy One more thing, after reviewing what you did, THIS NEW UPDATE IS AS GREAT AS ROCK AND ROLL! 🎸

Note:  Under contributions maybe we can put issue + pull request templates with some basic instructions to guide us for both the framework and documentation contributions by our community.  We can also explain how to contribute to the wiki once it is available for community use.

I put two "Docs" discussions up in Oqtane.Framework repo to reference myself with a proposed outline that can help for those who want a copy/paste solution or some sort of checklist that helps us maintain.  Once we refine this maybe it can be included as you submit issues and pull requests in GitHub like other repositories do?

**Comment by thabaum** [comment]
:rocket: A productive week for our Oqtane Docs community as today it looks like we gained a new contributor today and where able to merged some new material.  On top of that we had a tremendously productive meeting discussing how we can all improve to help contribute towards reaching our Oqtane Framework documentations goals.  

Some side notes that come to mind I would like to share:
1. It's OK and preferred to group any smaller commits such as spelling corrections into one larger pull request to avoid having to repeatedly created them.  It is to make it easier on our contributors and maintainer from having anything more to do than what is needed.  
2. Larger pull requests where you include content that relate to specific sections with added pages are very welcomed as well.  The TOC.yml file is the table of contents file located in each section generally if more than 2 pages.  Otherwise in the parent TOC.yml file.  It would be best to keep the pull request discussion focus on the specific documentation issue at hand if possible when dealing with anything beyond simple spelling/grammer/typo type of issues.
3. A new version selector which will switch from 5 to 5.1 allowing us to retain the older images and information while replacing what is currently out of date.
4. We all have our directions we are moving and through the meeting discussion were able to smooth out the road a bit ahead of us.
5. Last thing I want to point out is our focus and efforts will be 100% directed at 5.1 once released filling in all noticeable voids in current documentation.

It was a fun meeting all together there were a lot more things discussed than I could ever write all down here in a respectable amount of time.  I just felt like sharing a little insight for anyone who might wish to join the fun!

FYI:  The next meeting was moved up to the last Tuesday of next month which is the 27th of March.  There are currently 3 of us have made it to both our 2 consecutive meetings.  Lets see if we can make it 4++!  I believe we have some super strong synergies going right now that will continue to compound and build up our community in the days ahead.

Cheers!

**Comment by vnetonline** [comment]
@iJungleboy I would like to be part of this team. Can you please let me know how I can attend the monthly meetings?

**Comment by iJungleboy** [comment]
@vnetonline  sure - drop me a note on info 2sxc org and I'll add you to the teams invite.

**Comment by thabaum** [comment]
FYI Reminder this meeting was moved up one day just this month to accommodate schedules.  
Meeting is at Tuesday, March 26th at 8am PST/11am EST.

**Comment by thabaum** [comment]
@iJungleboy was this rescheduled this month for meeting today? or did we keep it set for tomorrow?

**Reply by iJungleboy** [reply]
it's for today - I'm in the meeting, but ATM all by myself...

**Comment by iJungleboy** [comment]
@thabaum @vnetonline @GregStevenson I'm in the meeting all by myself. 

Are you having trouble logging on, or not attending?

**Reply by thabaum** [reply]
@iJungleboy I am on my way

**Comment by vnetonline** [comment]
Dear team

Thanks for sending the meeting minutes, bit hard for me with my time zone the meeting is at 3am in the morning, I will review the minutes and will revert back .... definitely want to contribute whatever I can.

**Reply by thabaum** [reply]
@vnetonline sounds like a tough time to try to make.

**Reply by vnetonline** [reply]
> @vnetonline sounds like a tough time to try to make.  

@thabaum almost made it was up till 2am must have dozed off

**Reply by thabaum** [reply]
@vnetonline Thanks for trying! I put some more info below to help us engage as we can on.  Main thing is for all of us to get Visual Studio Code running with this project and this way our experiences are similar to help provide support to each other.  Follow the directions in our freshly enhanced installation manuals and any improvements there you see let us know with an issue and provide a PR to help us improve if you can.  This can include adding images to the topics by adding them to the assets folder.

I believe after that step is done and you can push changes to your own fork branches you can begin looking for areas to improve.  I am going to improve my process of this as well, and anything worth documenting on how to document we want to include.  However we do not want to teach DocFX, just what is expected of Oqtane Documentation in relation to DocFX to help anyone get started with a contribution.

If you have any repeated sections for example, we rename a file `_repeatedcontent.md` there is now an example use case of how to do this in the installation for the security sections.  This was something I learned today at the meeting @iJungleboy shared with us.  We can look to include helpful tips like these if not already included that we would like to expect contributors to learn on their own more.

Also I learned today that GitHub has a feature that was shown to us we can use to help showcase any doc enhancements by publishing the site itself and providing a link you can share.  We should document this as well as it is pretty helpful for sharing and when you are not familiar with it.

So anything we are discussing, unless someone is "on it", is open for community contributions by anyone that feels up to it.  Adding images to our installation processes that are smaller (not huge) files but decent enough to work with is one of our goals.  So if you go through a process of steps and want to put some images in be our guest :).  

If you have any questions please ask.  Starting up some discussions on how to contribute in any way important to our project contributions that helps with things like keeping things inline with expectations should help clear fog for others down the road as well.   

For example of things we can also look to document are things like using "Reset" Instead of "Resetting" in a section title to keep things shorter, more concise is another thing we should document as one of our contributor expectations that if you follow can help the maintainer not have to adjust after merging your PR.  Things like this help maintain the project, we need a section that includes our best practices.  We don't want to overkill this area and teach DocFX, but we do want to limit documentation contributor support costs.

Just tossing some ideas your way so you can get rolling with some PR's soon!

Happy to assist any way I can.

Thank you.

Cheers!

**Comment by thabaum** [comment]
Great meeting thanks @iJungleboy for waiting for me to get things figured out on my end.  Seems I had the wrong ID.  I will be sure to use and review the latest ID emailed to us as I had an issue using a wrong ID.  I don't think this information changes often so we should be OK here going forward.

🚀 **Meeting Highlights:**

- **Next Meeting Date:** April 30th, Tuesday, 8am EST. Make sure to use the latest ID provided via email.
  
- **Open Invitation:** Anyone interested in contributing to documentation efforts is welcome! Just shoot an email to @iJungleboy to join in.
  
- **Productive Meetings:** Our sessions are focused and fruitful, helping us shape documentation effectively.
  
- **Expanded Topics:** We've broadened our coverage to include more on installations, themes, modules, development and marketplace information.
  
- **Objectives for Next Meeting:** We aim to refine Maui Application Development, tidy up images, prep for the 5.1.0 release, and improve existing sections.
  
- **Tools Coverage:** We're enhancing documentation for common tools like Visual Studio and Visual Studio Code across Linux and Windows environments.
  
- **Newly Introduced:** A presentation slide is now part of the Upgrade topic, aiding in understanding the automated upgrade process.

📊 **Presentation Slides:** 

- **Upgrade Process:** Visual aids have been added to explain the automated upgrade process.  

---

Using presentation slides like in the `upgrade.md` file is and example of what we can use to help easily enhance other topics.  Using these slides presented in blogs, developer meetings and discussions/issues can help us visualize things.

---

Images we need to enhance down the road.   We also need to include a media page with Oqtane images for all use cases and marketing purposes to close #7.

---

5.1 will be a main focus with a thorough redo of the entire administration sections a priority following next weeks developer meeting.  We can try to utilize any slides @sbwalker provides to make issues or discussions on where and how they can be applied into our new documentation push over the next month.

---

The meeting notes have a lot more details. I believe we have a lot to chew on with the limited resources we currently work with, but collectively we should have documentation looking good by 5.1.1.  Maybe a version control of "Legacy (v5.0.2 and older)" and "Latest (v5.1.x)" is being considered to be added eventually as well. It sounds like next month is a busy month for everyone so this might be something to discuss as one of next months agenda topics.

Thanks again for everyone's help in all directions from reporting issues and starting discussions to assisting with pull requests and issues.  Eventually all the gears will be greased here.  Anyone providing feedback in any way helps us all find ways to contribute and improve documentation of the Oqtane Framework for everyone.  As the team grows with more active members [docs.oqtane.org](https://docs.oqtane.org ) documentation website will become a very interesting and helpful resource for Oqtane users to engage.

Cheers!

**Reply by GregStevenson** [reply]
Sorry, I missed the meeting. It did not show up in my calendar and I was just now checking because I thought it was Wed, or Thurs. 
I will go over the meeting notes.

--Greg

**Comment by vnetonline** [comment]
> I am wondering if it would be a good idea to have a late night for me, early morning for you "Documentation" meeting that maybe is on the flip side of the clock to the developer meeting and at the end of the month/ same day or whatever works best for @iJungleboy as our documentation lead.
> 
> 
> 
> It can be scheduled for 1 hour but maybe only takes half hour if that gets all agenda accomplished.
> 
> 
> 
> I think it would be a great place to help explain how to contribute, what we need and share ideas on improvements.   It can also serve as another meeting a developer that may not have been able to attend developer meetings due to time.  I would be able and willing to show up to 12 hours later time to the developers meeting on any day I don't have something come up.
> 
> 
> 
> This can be monthly, quarterly, semi-annual, annual or once per release scheduled out of the blue but with time to know about and prepare for. I just feel if there are any issues with contributors that need to get some things addressed but can't make it to a developer meeting it would give a place to catch up asking questions and documenting solutions to the answers!  We can probably work on it on the fly more as well resolving any issues that we can provide a quick solution for once a discussion makes it clear how to better serve that area.
> 
> 
> 
> I think a documentation meeting prior to the developer meeting would benefit getting issues discussed at the developer meeting, having the documentation meeting after the developer meeting would help provide a chance to help inform those who didn't make the developer meeting.
> 
> 
> 
> I believe some people are going to want to make their application from Oqtane, not as developers of modules, but more as users of modules developers produced both as products and services.  Which is what the Oqtane community marketplace developer ecosystem is after in the long run.  Creating information to help the end-users and developers supporting those users is the ultimate goal.
> 
> 
> 
> Not everyone that wants to use Oqtane will be a developer is what I am saying and I don't think we should limit Oqtane to only developers because that limits the Marketplace potential and attraction.  Some are more administrative types that want solutions developed and may choose their development team to use Oqtane based on how much is available to pull from.
> 
> 
> 
> This meeting can be on Discord for example has been mentioned to create a channel for this or on Teams like the developer meetings.  Maybe we can use that same room with a different password?
> 
> 
> 
> Some more food for thought on this one in the event you feel we are ready for this next step!  I think it can be more inclusive to anyone that wants to attend instead of only developers as a last note.  Ask everyone interested to use Oqtane.org contact form and have those requests forwarded to appropriate people.  
> 
> 
> 
> Maybe we can have a category on the form on that page to select to help direct messages immediately.  One for developer meetings request, documentation meeting requests and so on.  Might be good to show we have meetings to new prospects letting them know this project is active in these areas and create an events page with contact forms to fill out for each.
> 
> 
> 
> Just thinking out loud here a little, excited to see the new discussions forums!  Hopefully this first idea hits home, but I understand if it is a bit much at the same time.  I feel if this meeting needs canceled it is acceptable, as in not a mission critical part of Oqtane Framework.  No need to lose sleep over it... but I will for the meeting.

I can't seem to find the meeting minutes from the last meeting which occurred @iJungleboy can you please rend to me sorry for the trouble.

**Comment by iJungleboy** [comment]
Here are the minutes of the previous two meetings:

* [2024-02-28 Oqtane Docs - Minutes.pdf](https://github.com/oqtane/oqtane.docs/files/14831644/2024-02-28.Oqtane.Docs.-.Minutes.pdf)
* [2024-03-26 Oqtane Docs - Minutes.pdf](https://github.com/oqtane/oqtane.docs/files/14831648/2024-03-26.Oqtane.Docs.-.Minutes.pdf)

**Comment by thabaum** [comment]
@iJungleboy thanks for your time holding today's meeting for the 3 of us who attended (including you).

This month as per our discussion I will get on the Maui developing and helpful images along with any corrections needed to install and setup instructions in the coming weeks for review.

Hopefully we can get another 1 or 2 of us pushing up some PR's. 

500+ forks we should have at least 1% (5) people working on documenting things by now.  Hopefully we can gain another 1 or 2 active contributors here to help make things more interesting for this project and our meetings.

Talk more tomorrow at the developer meeting.

🚀

**Comment by thabaum** [comment]
Great meeting today, short, sweet and to the point... need to get some stuff done beside what our maintainer has provided.  I said to myself yesterday I was going to get something done prior to today's meeting and failed to do so.  In failing my own goals set I promise to put some contributions on the table this week starting today with issue #65 and later issue #60 as I mentioned to @iJungleboy.

Rock and roll 🎸 here I go... I need to get this stuff off my plate.

On another awesome note 🎵 I love the recent updates made to our docs and would like to **thank @iJungleboy** for making the "Guides" section extremely nice... wow.  Took a minute due to the changes to set in, but once I took it from a first time viewing our docs perspective things are really starting to flow.  Nice work!  I am sure I will find more as I go through things again as I have to admit it has been a while.

Time for me to catch up I will see what I can do as I have been meaning to fire up Visual Studio Code which @iJungleboy created a great section in the "[Tips](https://docs.oqtane.org/guides/docs/tips/index.html)" page for the contributing section.

I will have #65 done in couple hours and report any updates on the [contributer tips](https://docs.oqtane.org/guides/docs/tips/index.html) section.  Then I may dive in into #60 who knows have some of it done by end of day well.

I have to review latest commits but looks like things are moving forward here nicely and methodically as discussed.  Keep on keeping on!

I seen we had a few new contributors as well along the way... thank you all who have helped with small and big documentation issues like spelling and finding things that may be outdated or incorrect.  Together we are making things nicer for the next person that visits our Oqtane community documentation home. 

# 😎

**Comment by thabaum** [comment]
I will be at next meeting and since it is fall/winter time I should be able to clear all those issues I created by the years end.  I will add some notes for installations disabling WebDAV in control panels as I seen a few people had issues since posting the installation instructions.

**Reply by iJungleboy** [reply]
great! thx

**Reply by thabaum** [reply]
@iJungleboy and @GregStevenson I created some additional issues along with updating clarification on where they should be included.  

I also created my first PR here in a while as promised.  

With these new issues and updates to documentation requests we should have something worth discussing at our upcoming meeting.

Thank you!

**Reply by thabaum** [reply]
Is our meeting today or next week?  My calendar is popping up like we have it today but maybe it is set recurring to 4th Tuesday instead of last.  I believe it is the last Tuesday 8am PST correct?

**Reply by thabaum** [reply]
I believe our meeting is set to the last Tuesday of each month, but we adjusted it one month which threw my calendar off.  I will set this for the last month again.  We get an extra Tuesday this month so it is making our date/time off on the recurring.

**Reply by iJungleboy** [reply]
yes, next week.
